app/controllers/metrics_controller.rb

```diff
@@ -0,0 +1,17 @@
+class MetricsController < ApplicationController
+  skip_before_action :set_sentry_context, only: [:index]
+  skip_before_action :check_locale, only: [:index]
+  skip_around_action :switch_locale, only: [:index]
+
+  before_action :authenticate
+
+  def index
+    @metrics = MetricsReport.new
+  end
+
+  def authenticate
+    authenticate_or_request_with_http_basic do |username, password|
+      username == ENV['METRICS_USERNAME'] && password == ENV['METRICS_PASSWORD']
```

Comment:
If these ENV vars aren't set, would the auth modal pop up but it'd be looking for blank strings? Maybe first have `return if ENV['METRICS_USERNAME'].blank?` like in the concern.

Reply:
I thought so but apparently not. I realized the test I wrote still assumed the vars were set. But `return`ing when it's blank just lets it continue without authorization. I'll need to look at this further.

Reply:
I THINK it works without the return. username/pw when not set by the header are nil, and ENV[stuff], when not set, is nil, so we get `nil == nil && nil == nil` - which...is true.

I put the return in there mostly to clarify what I was doing to begin with (I find it easier to read and to follow).